src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Windows.targets

```diff
@@ -167,4 +167,72 @@ Copyright (c) .NET Foundation. All rights reserved.
                       and '$(UseUwp)' == 'true' ">
     <NetSdkError ResourceName="WindowsSDKXamlInvalidTfm" />
   </Target>
+
+  <!--
+    ============================================================
+                        _RunCsWinRTGenerator
+
+    Runs 'cswinrtgen' to produce the 'WinRT.Interop.dll' assembly.
+    ============================================================
+  -->
+  <UsingTask Condition="'$(CsWinRTGeneratorTasksOverriden)' != 'true'" TaskName="RunCsWinRTGenerator" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target
```

Comment:
Ideally, this target would only run if something changed. As in, if there's been any code changes or if any transitive dependency changed. I'm assuming the `Csc` task that's invoked in `CoreCompile` would already be doing some kind of check like that, to skip work if eg. you just F5 again without changing anything. Is there a way for us to perform the same check (or get that result out of `CoreCompile`) so we can also skip this target if not needed?

Reply:
This concept is known as [_incrementality_](https://learn.microsoft.com/en-us/visualstudio/msbuild/incremental-builds?view=vs-2022). The short form is that if you know your inputs, and your inputs are files whose timestamps can be tracked, then you can set `Inputs=..all that stuff..` on your `Target` and the MSBuild engine will transparently track them and only call your target if any of them are out of date. Similarly, if you know your Outputs, you can provide those to the engine as well and up-to-date checking will take place.

This means that many complex Targets often have a 'compute the inputs and outputs for Target XXXXX' Target that runs immediately before them, so that you can compute the correct set of inputs/outputs for incrementality purposes.

Reply:
Thank you! I added the `Inputs` property. I have a couple more questions though:
- I don't know the exact output path in advance, as that's computed by the target. I get that via that `CsWinRTGeneratorInteropAssemblyPath` item that the target produces as output. Is there a way to configure that returned item as the `Outputs` item for the wrapping task?
- I also need to re-run the target if any of the other input properties changes. The docs seem to only talk about input/output items. How can one handle incrementality with respect to properties as well? 🤔
- 
Thank you!

Reply:
For the first point, we usually try to have those paths be able to be _input_ into the Task to help with this kind of incrementality.
For the second, this is a gap in the MSBuild Incrementality model - it only supports _files_ today. To get around this, most people have a 'generate property cache file' Target that they have run immediately before their real Target and they add this generated file to their Inputs. You can see an example of this pattern [here](https://github.com/dotnet/sdk/blob/897c950057bc8157ad7a7aac77d04fec3ed23920/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L1088-L1112).

Reply:
That link was super helpful, thank you! I've made some changes following that reference 🙂